shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -868,10 +869,121 @@ - (instancetype)initWithOwner:(FlutterTextInputPlugin*)textInputPlugin {
   return self;
 }
 
+- (void)handleSearchWebAction {
+  [self.textInputDelegate flutterTextInputView:self
+                     searchWebWithSelectedText:[self textInRange:_selectedTextRange]];
+}
+
+- (void)handleLookUpAction {
+  [self.textInputDelegate flutterTextInputView:self
+                            lookUpSelectedText:[self textInRange:_selectedTextRange]];
+}
+
+- (void)handleShareAction {
+  [self.textInputDelegate flutterTextInputView:self
+                             shareSelectedText:[self textInRange:_selectedTextRange]];
+}
+
+// DFS algorithm to search a UICommand from the menu tree.
+- (UICommand*)searchCommandWithSelector:(SEL)selector
+                                element:(UIMenuElement*)element API_AVAILABLE(ios(16.0)) {
```

Comment:
Is this searching through the default menu (suggestedActions) trying to find the `item` given in the call to showSystemContextMenu? Won't all of the ones that we currently support never be nested? Or maybe I just don't understand how suggestedActions is structured.

Reply:
Actually all of them are nested at least 1 level deep in my observation - Apple uses [`UIMenuOptionsDisplayInline`](https://developer.apple.com/documentation/uikit/uimenuoptions/uimenuoptionsdisplayinline?language=objc) to make the items appear on the top level. And since this is not an API contract, we have to do a tree search here.

Reply:
Got it, thanks for the explanation!